src/lib/api.js

```diff
@@ -0,0 +1,92 @@
+import {DeployKey} from "./deploy-key.js";
+import {zip} from "zip-a-folder";
+import fs from "fs";
+import {UnknownError} from "./errors.js";
+import parse from "./resource.js";
+import { tmpdir } from "node:os";
+import {sep} from "node:path";
+import {printWhile} from "./cli.js";
+
+export async function apiDeploy(key, deployFolder, releaseNote, {stdin, stdout, stderr}, config) {
```

Comment:
I think this is fine as-is for this commit, but passing in `stdin` _et al._ pierces the abstraction a little by coupling this to the command line, which may not be as useful when reusing this for the GitHub action. Perhaps we need a function to build the zip and another function to upload it, then the `printWhile`s could be in the `deploy` command.

Reply:
Decided to postpone this part until have more info regarding github actions